addons/stock_account/models/product.py

```diff
@@ -235,17 +235,48 @@ def _change_standard_price(self, new_price, counterpart_account_id=False):
 
         # Actually update the standard price.
         self.with_context(force_company=company_id.id).sudo().write({'standard_price': new_price})
+        
+    def _get_fifo_svl_candidates(self, quantity, company, lots=None):
+        """
+        Find back incoming stock valuation layers (called candidates here). If lots are given, this will find
+        the first incomming stock valuation layers associatted with the lots before finding further.
 
-    def _run_fifo(self, quantity, company):
+        :param quantity: the quantity of the first incoming stock valuation layers
+        :param company: the corresponding company of the operation
+        :param lots: if passed, force finding back the the first incoming stock valuation layers corresponding the the given lots/serials recordset
+        :return recordset of stock.valuation.layer of the product that are associated with the given lots (if lots is given) and have positive remaining quantity
+        """
+        self.ensure_one()
+        SVL = self.env['stock.valuation.layer']
```

Comment:
user tiny
`valuation_layer = self.env['stock.valuation.layer']`

Reply:
Agree. Which one is better between `ValuationLayer` and `valuation_layer`?

Reply:
`valuation_later`